db/versions/0017.yml

```diff
@@ -102,3 +102,63 @@ methods:
           end;
         end if;
       end
+  add_denylist_address:
+    description: Add a denylist address for the taskcluster-notify service.
+    mode: write
+    serviceName: notify
+    args: address_in jsonb
+    returns: void
+    body: |-
+      begin
+        insert into denylisted_notification(notification_type, notification_address, etag)
+        values (
+          (address_in ->> 'notificationType')::text,
+          (address_in ->> 'notificationAddress')::text,
+          public.gen_random_uuid()
+        ) on conflict do nothing;
```

Comment:
If there is a conflict, it means an entry being has the same notification address and notification type, but a different etag. Since there are no other fields, keeping the existing entry is sufficient, and therefore no change is required. Since the operation should be idempotent, we can avoid throwing an error.

Reply:
From what I can see, the source code expects an error to be thrown when the row already exists. It's surrounded by a try catch statement and currently checks for `EntityAlreadyExistsError`.

Reply:
True, but in that case it just returns -- the overall effect of the SQL + JS is to ignore conflicts.  This is the idempotency that Pete refers to above.  I think implementing that idempotency at the DB level is fine, and means that the try/catch can be removed in the JS code that calls the DB function.

Reply:
It's worth putting a sentence in the function's `description` to indicate that, though.